test/e2e/simple_test.go

```diff
@@ -64,12 +63,16 @@ func TestE2ESimple(t *testing.T) {
 	err = txsim.Run(ctx, testnet.GRPCEndpoints()[0], kr, encCfg, opts, sequences...)
 	require.True(t, errors.Is(err, context.DeadlineExceeded), err.Error())
 
+	expectedAppVersion, err := testnode.ReadAppVersionFromGenesis(context.Background(),
+		testnet.Node(1).AddressRPC())
+	require.NoError(t, err)
+
 	blockchain, err := testnode.ReadBlockchain(context.Background(), testnet.Node(0).AddressRPC())
 	require.NoError(t, err)
 
 	totalTxs := 0
 	for _, block := range blockchain {
-		require.Equal(t, v1.Version, block.Version.App)
+		require.Equal(t, expectedAppVersion, block.Version.App)
```

Comment:
[question] do we run this e2e test on the v1.x branch? If not, it seems like we could've done this:

```diff
-		require.Equal(t, v1.Version, block.Version.App)
+		require.Equal(t, v2.Version, block.Version.App)
```

but your approach seems more resilient to future app versions so seems preferable to me.